pkg/e2e/verify/nodepool.go

```diff
@@ -0,0 +1,474 @@
+	"github.com/openshift/osde2e/pkg/common/config"
+	"github.com/openshift/osde2e/pkg/common/expect"
+	"github.com/openshift/osde2e/pkg/common/helper"
+	"github.com/openshift/osde2e/pkg/common/label"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"sigs.k8s.io/e2e-framework/klient/k8s/resources"
+	"sigs.k8s.io/e2e-framework/klient/wait"
+	"sigs.k8s.io/e2e-framework/klient/wait/conditions"
+)
+
+const (
+	nodePoolLabel = "hypershift.openshift.io/nodePool"
+	awsProvider   = "aws"
+)
+
+var nodePoolGVR = schema.GroupVersionResource{
+	Group:    "hypershift.openshift.io",
+	Version:  "v1beta1",
+	Resource: "nodepools",
+}
+
+var _ = ginkgo.Describe("[Suite: e2e] NodePool STS Permissions", ginkgo.Ordered, label.HyperShift, label.E2E, func() {
+	var (
+		h                *helper.H
+		client           *resources.Resources
+		clusterNamespace string
+		testNodePoolName string
+		createdNodePool  bool
+		canAccessAPI     bool
+	)
+
+	ginkgo.BeforeAll(func() {
+		cloudProvider := viper.GetString(config.CloudProvider.CloudProviderID)
+		if cloudProvider != awsProvider {
+			ginkgo.Skip(fmt.Sprintf("Tests only supported on AWS, got %s", cloudProvider))
+		}
+
+		h = helper.New()
+		client = h.AsUser("")
+
+		var err error
+		clusterNamespace, err = getClusterNamespace(h)
+		if err != nil {
+			ginkgo.Skip(fmt.Sprintf("Cannot determine cluster namespace: %v", err))
+		}
+
+		canAccessAPI = canAccessNodePoolAPI(h, clusterNamespace)
+		testNodePoolName = fmt.Sprintf("sts-test-%d", time.Now().Unix()%100000)
+	})
+
+	ginkgo.AfterAll(func() {
+		if createdNodePool {
+			err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).
+				Delete(context.Background(), testNodePoolName, metav1.DeleteOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				ginkgo.GinkgoLogr.Error(err, "Failed to cleanup test NodePool", "name", testNodePoolName)
+			}
+
+			wait.For(func(ctx context.Context) (bool, error) {
+				_, err := h.Dynamic().Resource(nodePoolGVR).Namespace(clusterNamespace).
+					Get(ctx, testNodePoolName, metav1.GetOptions{})
+				return apierrors.IsNotFound(err), nil
+			}, wait.WithTimeout(3*time.Minute), wait.WithInterval(10*time.Second))
+		}
+	})
+
+	ginkgo.It("validates existing nodes have proper AWS integration", func(ctx context.Context) {
+		nodePoolNodes, err := findNodePoolNodes(ctx, client)
+		failIfSTSError(err, "node discovery", "ec2:DescribeInstances")
+		expect.NoError(err, "Failed to find NodePool nodes")
+		Expect(len(nodePoolNodes)).To(BeNumerically(">", 0), "No NodePool nodes found")
+
+		for _, node := range nodePoolNodes {
+			validateAWSIntegration(node)
+		}
+	})
+
+	ginkgo.Describe("NodePool creation and provisioning", func() {
+		ginkgo.It("creates new NodePool successfully", func(ctx context.Context) {
```

Comment:
Wouldn't you want to create the nodepool in a `BeforeAll` and then you can get rid of the `createdNodePool` boolean?